plugins/automation/spec/lib/scriptable_spec.rb

```diff
@@ -213,6 +213,81 @@
       end
     end
 
+    describe ".build_quote" do
+      fab!(:user) { Fabricate(:user, name: "John Doe", username: "johndoe") }
+      fab!(:post) { Fabricate(:post, user: user, raw: "This is a post content", post_number: 1) }
+
+      before do
+        SiteSetting.display_name_on_posts = false
+        SiteSetting.prioritize_username_in_ux = false
+      end
+
+      context "when post is nil" do
+        it "returns an empty string" do
+          expect(DiscourseAutomation::Scriptable::Utils.build_quote(nil)).to eq("")
```

Comment:
We could probably extract `DiscourseAutomation::Scriptable::Utils.build_quote(post)` to `subject(:quote)`.